src/components/organisms/AuthenticationModal/RegisterForm/RegisterForm.tsx

```diff
@@ -89,6 +89,28 @@ const RegisterForm: React.FunctionComponent<PropsType> = ({
   const onSubmit = async (data: RegisterFormData) => {
     try {
       setShowLoginModal(false);
+
+      if (venue.access === VenueAccessMode.Emails) {
+        const AccessEmails = await firebase
+          .firestore()
+          .collection("venues")
+          .doc(venue.id)
+          .collection("access")
+          .doc("Emails")
+          .get();
+
+        const emails = AccessEmails.data();
+
+        if (!emails?.emails.includes(data.email.toLowerCase())) {
+          setError(
+            "email",
+            "validation",
+            "We can't find you! Please use the email from your invitation."
+          );
+          return;
+        }
+      }
```

Comment:
Why was this implemented on the frontend like this?

There was code for checking emails in the original Venue Access PR, using the secure backend function designed for this purpose:

- https://github.com/sparkletown/sparkle/pull/1016/files?file-filters%5B%5D=.tsx#diff-bf148459f99d12aaa5ffbfa3bd6c38c22e596f8a0dfe39361093e57d6d585596

It looks like this implementation was removed in https://github.com/sparkletown/sparkle/pull/1159, though I didn't read deeply into the specifics of why/how that applies to the use case here.